src/v/config/convert.h

```diff
@@ -360,4 +360,34 @@ struct convert<model::cloud_storage_backend> {
     }
 };
 
+template<>
+struct convert<model::leader_balancer_mode> {
+    using type = model::leader_balancer_mode;
+
+    static constexpr auto acceptable_values = std::to_array(
+      {"random_hill_climbing", "greedy_balanced_shards"});
+
+    static Node encode(const type& rhs) { return Node(fmt::format("{}", rhs)); }
+
+    static bool decode(const Node& node, type& rhs) {
+        auto value = node.as<std::string>();
+
+        if (
+          std::find(acceptable_values.begin(), acceptable_values.end(), value)
+          == acceptable_values.end()) {
+            return false;
+        }
+
+        rhs = string_switch<type>(std::string_view{value})
+                .match(
+                  "random_hill_climbing",
+                  model::leader_balancer_mode::random_hill_climbing)
+                .match(
+                  "greedy_balanced_shards",
+                  model::leader_balancer_mode::greedy_balanced_shards);
+
+        return true;
+    }
```

Comment:
There are multiple copied defined for these string literals. It may be beneficial to reduce them to just one place knowledgeable of string representation of the values, and reuse it. That way someone intoducing new modes in future will not forget to update one of those places.

Introducing a function like `bool string_switch::has_result()` can help getting rid of `acceptable_values` in particular.

Reply:
Switching to only defining the string literal conversion in a single location. Definitely could reduce user error in the future.